src/_pytest/nodes.py

```diff
@@ -448,17 +449,9 @@ def add_report_section(self, when: str, key: str, content: str) -> None:
     def reportinfo(self) -> Tuple[str, Optional[int], str]:
         return self.fspath, None, ""
 
-    @property
+    @cached_property
     def location(self) -> Tuple[str, Optional[int], str]:
-        try:
-            return self._location
-        except AttributeError:
-            location = self.reportinfo()
-            fspath = self.session._node_location_to_relpath(location[0])
-            assert type(location[2]) is str
-            self._location = (
-                fspath,
-                location[1],
-                location[2],
-            )  # type: Tuple[str, Optional[int], str]
-            return self._location
+        location = self.reportinfo()
```

Comment:
With regard to this I've noticed that we'd want to cache `reportinfo()` probably/maybe - any ideas how to do this?  Should/could we wrap it in a property with a different name then?  `lru` might also be an option, but seems bad with just a single item?!

Reply:
I think for methods we can implement a simple decorator that does something similar to `cached_property`, say `cached_method`. It should be even simpler than `cached_property`.

Reply:
>  lru might also be an option, but seems bad with just a single item?!

lru_cache is very fast in my experience (in Python terms...). However, it can't really be used on methods, it leaks `self`.

Reply:
lets leave the method caching for later for now,
it can trivially be implemented as shim for a new private  property if necessary